src/tests/JIT/HardwareIntrinsics/Arm/Shared/Helpers.cs

```diff
@@ -8717,14 +8727,27 @@ public static ulong Splice(ulong[] first, ulong[] second, ulong[] maskArray, int
                 return false;
             }
 
-            var validElementCount = firstOp.Length;
+            var validElementCount = data.Length;
+            var hasFaulted = false;
             var expectedFaultResult = 
                 InitVector<TFault>(i =>
                 {
+                    if (hasFaulted)
+                    {
+                        return TFault.Zero;
+                    }
+
+                    if (mask[i] == TElem.Zero)
+                    {
+                        return TFault.One;
```

Comment:
if `mask[i] == 0`, shouldn't this lane should also be `TFault.Zero`?

Reply:
This part calculates the expected FFR value.

> Inactive elements will not cause a read from Device memory or signal a fault, and are set to zero in the destination vector.

Since there's no fault, the FFR element is `1`. You can check the pseudo code here: https://docsmirror.github.io/A64/2023-06/ldff1b_z_p_br.html

Reply:
ah, thanks for pointing it out.